### PR TITLE
As a developer, I want the default DGPF styling to be a new V3 [sc-32980]

### PR DESCRIPTION
**Pull request by Sudershan29**
1. I copied the templates from templates/v2, and copied styling from the new USGS portal based on Vytas's suggestion.
2. I created a v3 directories for both templates and css
3. I made sure to check reverse compatibility by changing BASE_TEMPLATE from v2 to v3, and v3 back to v2.

### CODE REVIEW

**Comment by NickolausDS** [comment]
@Sudershan29 Since these are UI changes, could you add a few screenshots to show what the new changes look like?

Also, did you find a way for the breadcrumb feature to work within the new templates?

File: globus_portal_framework/settings.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def get_rfm(search_result):
 
 ROOT_URLCONF = 'globus_portal_framework.urls'
 
-BASE_TEMPLATES = 'globus-portal-framework/v2/'
+BASE_TEMPLATES = 'globus-portal-framework/v3/'
```
Comment (NickolausDS) [inline globus_portal_framework/settings.py:185]: Could we leave these as v2 for now? New folks can switch to v3 manually for v0.4.x, and once we're ready to roll out v0.5.x we can make this default.

File: globus_portal_framework/static/css/v3/search.css [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+    border-right-width: 0px;
+    border-top-width: 0px;
+    border-bottom-width: 0px;
+    padding-bottom: 0;
+}
+
+#details .card {
+    border: 0;
+    padding-bottom: 25px;
+}
```
Comment (NickolausDS) [inline globus_portal_framework/static/css/v3/search.css:321]: Like the other static files, could we move this under the `globus-portal-framework` folder?

File: globus_portal_framework/templatetags/random_numbers.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+from django import template
+
+register = template.Library()
+
+
+@register.simple_tag
+def random_int(a, b=None):
+    if b is None:
+        a, b = 0, a
+    return random.randint(a, b)
```
Comment (NickolausDS) [inline globus_portal_framework/templatetags/random_numbers.py:11]: What is this for?

File: globus_portal_framework/templates/globus-portal-framework/v3/components/search-nav.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{% load static %}
+
+<div class="subnav mt-auto bottom-border-0">
+  <ul class="nav nav-tabs nav-search-nav justify-content-center">
+    <li class="mt-auto">
+      <a class="btn btn-primary btn-lg border-0 squared {% if active == 'search' %}active{% endif %}" role="button"
+        href="{% url 'search' globus_portal_framework.index %}">Search</a>
+    </li>
+    <li class="mt-auto">
+      <a class="btn btn-primary btn-lg border-0 squared {% if active == 'about' %}active{% endif %}" role="button"
```
Comment (NickolausDS) [inline globus_portal_framework/templates/globus-portal-framework/v3/components/search-nav.html:10]: You should be able to do this with the `is_active` template tag in DGPF. And then you won't need to pass context and manually check this. There should be examples in the other templates, it should look something like this: 

```
<li class="breadcrumb-item {% is_active request 'search-about' index=globus_portal_framework.index %}">
```

**Comment by NickolausDS** [comment]
There are great changes! One other thing that would be really useful would be a migration guide for someone upgrading from the v2 to v3 templates. Specifically if anything they were using in V2 would not work in V3. 

A migration doc, much the same way we have in [gladier here], would be a good place to put that.

File: globus_portal_framework/settings.py (hunk 1 above)
Comment (Sudershan29) [reply globus_portal_framework/settings.py:185]: Sounds good!

File: globus_portal_framework/templatetags/random_numbers.py (hunk 3 above)
Comment (Sudershan29) [reply globus_portal_framework/templatetags/random_numbers.py:11]: They had in the USGS project, and I assumed it was getting called on the HTML page. But you are right, looks like its doing nothing on the `search-results.html` page. Ill remove the file and the reference on that page.

**Comment by Sudershan29** [comment]
This is how the new DGPF would look like with v3 versioning

DGPF-V3
DGPF-V3-Page2

**Comment by NickolausDS** [comment]
I looked over the changes locally and everything looks good! There may still be some tweaks before the v0.5.0 release where this becomes default, but this is a good starting point for now.

Thanks @Sudershan29, nice work!

**Pull request merged**